crates/electrum/tests/test_electrum.rs

```diff
@@ -190,3 +193,57 @@ fn tx_can_become_unconfirmed_after_reorg() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_last_seen_unconfirmed() -> Result<()> {
+    use bdk_chain::ConfirmationHeightAnchor;
+    let env = TestEnv::new()?;
+    let client = electrum_client::Client::new(env.electrsd.electrum_url.as_str())?;
+
+    // Setup receiver
+    let addr =
+        Address::from_str("bcrt1qj9f7r8r3p2y0sqf4r3r62qysmkuh0fzep473d2ar7rcz64wqvhssjgf0z4")?
+            .assume_checked();
+    let spk = addr.script_pubkey();
+    let (chain, _) = LocalChain::from_genesis_hash(genesis_block(Network::Regtest).block_hash());
+    let mut graph = {
+        let mut index = SpkTxOutIndex::<u32>::default();
+        index.insert_spk(0, addr.script_pubkey());
+        IndexedTxGraph::<ConfirmationHeightAnchor, _>::new(index)
+    };
+
+    // Mine blocks
+    let _ = env.mine_blocks(101, None)?;
+
+    // Send tx and confirm it
+    let amt = Amount::from_sat(10_000);
+    let txid = env.send(&addr, amt)?;
+    let _ = env.mine_blocks(1, None)?;
+
+    // Sync. Last seen time should be 0 for confirmed tx
+    env.wait_until_electrum_sees_block()?;
+    let ElectrumUpdate { relevant_txids, .. } =
+        client.sync(chain.tip(), [spk.clone()], None, None, 5)?;
+    let missing = relevant_txids.missing_full_txs(graph.graph());
+    let graph_update = relevant_txids.into_tx_graph(&client, None, missing)?;
+    let _ = graph.apply_update(graph_update);
+    let tx = graph.graph().full_txs().next().unwrap();
+    assert_eq!(tx.txid, txid);
+    assert_eq!(tx.last_seen_unconfirmed, 0);
+
+    // Send tx 2
+    let txid = env.send(&addr, amt)?;
+
+    // Mine empty block and sync. TxGraph should record last seen unconfirmed
+    let _ = env.mine_empty_block();
+    env.wait_until_electrum_sees_block()?;
+    let ElectrumUpdate { relevant_txids, .. } = client.sync(chain.tip(), [spk], None, None, 5)?;
+    let missing = relevant_txids.missing_full_txs(graph.graph());
```

Comment:
we want to avoid writing these kinds of tests that use `apply_update` and `missing_full_txs` because we're going to change the mechanisms on how this works. Tests against the blockchain backends should just check that it returns the right data (not that it is right after applying it as an update).

Reply:
You're right, it's a bit overkill.